metals/src/main/scala/scala/meta/internal/metals/onTypeRangeFormatters/OnTypeRangeFormatter.scala

```diff
@@ -0,0 +1,36 @@
+package scala.meta.internal.metals.onTypeRangeFormatters
+
+import scala.meta.tokens.Tokens
+
+import org.eclipse.lsp4j.FormattingOptions
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+
+trait OnTypeRangeFormatter {
+  type StartPosition = meta.Position
+  type EndPosition = meta.Position
+
+  def onRangeContribute(
+      sourceText: String,
+      range: Range,
+      splitLines: Array[String],
+      startPos: StartPosition,
```

Comment:
How is `startPos` and `endPos` different from range?

Reply:
TBF, they were already implemented like this in the previous `MultilineStringFormattingProvider` implementation. 
As far as I can understand the `startPos` and `endPos` returned from the `withToken` method are the starting and ending position in the virtual file used in the `withToken` method. I'm not sure they actually change.

Reply:
I get that, but since we are doing an API for more formatters, it might be a time to tidy it up. So if possible let's reduce the number of parameters, we can create an additional case class to maybe contain them if it's not possible to throw some out.